File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
from collections import deque

input = sys.stdin.readline

V = int(input())
E = int(input())
graph = [[] for _ in range(V + 1)]
for _ in range(E):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

visited = [False] * (V + 1)


# bfs 풀이
def bfs(start):
    Q = deque([start])
    visited[start] = True
    while Q:
        v = Q.popleft()
        for i in graph[v]:
            if not visited[i]:
                Q.append(i)
                visited[i] = True


bfs(1)
print(visited.count(True) - 1)

File: 백준/Silver/2606. 바이러스/바이러스.js
// let input = require('fs').readFileSync('2606.txt').toString().split('\n');
let input = require('fs').readFileSync('dev/stdin').toString().split('\n');

const node = Number(input[0]);
const edge = Number(input[1]);

// graph 초기화
let graph = [];
for (let i = 0; i <= node; i++) {
  graph.push([]);
}

// 각 노드의 관계 정리
for (let i = 0; i < edge; i++) {
  const [a, b] = input[i + 2].split(' ').map((v) => Number(v));
  graph[a].push(b);
  graph[b].push(a);
}

let visited_bfs = new Array(node + 1).fill(false);
let answer_bfs = [];
function bfs(graph, start) {
  let queue = [start];
  visited_bfs[start] = true;
  answer_bfs.push(start);
  while (queue.length) {
    let v = queue.shift();
    for (let i = 0; i < graph[v].length; i++) {
      if (visited_bfs[graph[v][i]] === false) {
        queue.push(graph[v][i]);
        visited_bfs[graph[v][i]] = true;
        answer_bfs.push(graph[v][i]);
      }
    }
  }
}
bfs(graph, 1);
console.log(answer_bfs.length - 1);
